### ISSUE TITLE
How is this teaching material?

### ISSUE THREAD

**Issue by felixsanz** [timoxley/functional-javascript-workshop#149]
I'm enjoying this workshop, but i'm also having a really hard time trying to understand the official solutions and also the exercise summaries. For example:

> Duck typing is a style of dynamic typing in which an object's methods and properties determine the valid semantics, rather than its inheritance from a particular class or implementation of a specific interface.

Really? Can someone explain this as if i were a duck, please?

**Comment by nprz** [comment]
This is probably too late but whateves :P 
I don't know if you have experience with something like C++, but in C++ once you create a class and then make an object from that class, all the methods and members for that class are set. So once an object has been created its type is that class. No confusion. 

This is different from JavaScript since you can create an object from a constructor (also called a prototype in JavaScript) and then call or add a method on that object which is NOT in that objects constructor/prototype. So now this object is that object + new method. 

So basically, an object in JavaScript may or may not have a certain method or member. And if the object does not have certain members or methods then we may not be able to call some other function on the object. So we can check to see if the object has this property with object.hasOwnProperty. 

Normally in an object oriented language you know all the methods that can be called on an object, in JavaScript you don't. 

I think that's all correct someone feel free to add or correct if any mistakes were made
